### PR TITLE
Disable User in Microsoft Active Directory with Microsoft Teams

### PR DESCRIPTION
**Pull request by mhofert-r7**
## Requirements

Workflow contributors, verify you have completed the following items by checking them off:

### Checklist
- [ ] Workflow verified as working in a test
- [ ] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
- [ ] Workflow title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Workflow includes a detailed Description
- [ ] Workflow step names follow [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Triggers, Decisions, and other workflow steps include descriptions where needed
- [ ] Workflow includes at least one `Final Report` artifact that summarizes workflow execution details
- [ ] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_of_Workflow/`
- [ ] `help.md` file includes Description, Key Features, Requirements, Setup, Technical Details, Version History, and References
- [ ] `workflow.spec.yaml` file includes `products`, `name`, `title`, `description`, `use_cases` tags, `keywords` tags, `source_url`, and the `name` for each screenshot matches a PNG image in the `screenshots` directory

### Assessment

Post your results with screenshots in this PR using the following style:

[*] Validating workflow at .

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Workflow successfully validated!

----
[*] Total time elapsed: 24.693ms

### Screenshots

Screenshots help your reviewers provide valuable feedback on your contribution. To add a screenshot here, simply drag and drop the PNG files from a file explorer.

#### Workflow Builder

workflow1
workflow2
workflow3

#### teams
teams1
teams2

#### Artifact

artifact1

### CODE REVIEW

File: workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/help.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+Disable User in Microsoft Active Directory with Microsoft Teams.
+
+Sample Slack Trigger Commands:
+
+`!disable_user <exampleuser>`
+
+# Key Features
+
+* Suspend an Active Directory LDAP user directly from Microsoft Teams
```
Comment (mrinehart-r7) [inline workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/help.md:11]:
```suggestion
* Disable an Active Directory LDAP user directly from Microsoft Teams
```

File: workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/help.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+* [Microsoft Teams](https://insightconnect.help.rapid7.com/docs/microsoft-teams)
+* Active Directory LDAP connection
+
+# Documentation
+
+## Setup
+
+Import the workflow from the Rapid7 Extension Library and proceed through the Import Workflow wizard in InsightConnect. Import plugins, create or select connections, and rename the workflow as a part of the Import Workflow wizard as necessary.
+Once the workflow has been imported, each Microsoft Teams step will need the team name and channel name updated to suit your Teams environment (edit the input with the preset text of `change_me`).
```
Comment (mrinehart-r7) [inline workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/help.md:23]:
```suggestion

Once the workflow has been imported, each Microsoft Teams step will need the team name and channel name updated to suit your Teams environment (edit the input with the preset text of `change_me`).
```

File: workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/help.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+To run the workflow, send the command `!disable_user <exampleuser>` in your configured team and channel. The workflow will post after it has completed.
```
Comment (mrinehart-r7) [inline workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/help.md:24]:
```suggestion

To run the workflow, send the command `!disable_user <exampleuser>` in your configured team and channel. The workflow will post after it has completed.
```

File: workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/help.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+In the Team and chanel inputs replace `change_me` with the appropriate team and channel.
+
+To run the workflow, in the configured channel enter the command `!disable_user <exampleuser>`.
+Example: `!disable_user jdoe`
+The workflow will acknowledge the request and reply when it has
+completed.
```
Comment (mrinehart-r7) [inline workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/help.md:30]: Remove these lines, they're covered by the paragraphs above.

File: workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/workflow.spec.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
+extension: workflow
+products: ["insightconnect"]
+name: Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams
+title: "Disable User in Microsoft Active Directory with Microsoft Teams"
+description: Suspend Active Directory LDAP users from Slack.
```
Comment (mrinehart-r7) [inline workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/workflow.spec.yaml:5]:
```suggestion
description: Disable Active Directory LDAP users from Slack.
```

File: workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams.icon [hunk 6]
```diff
@@ -681,7 +681,7 @@
             "parameters": {
               "input": {
                 "channel_name": "change_me",
-                "message_content": "⚠️ User '{{[a4934e67-ceaa-4eda-9b1f-c023175b37e9].[user_name]}}' was not found",
+                "message_content": "⚠️ Disable Active Directory User request for {{[a4934e67-ceaa-4eda-9b1f-c023175b37e9].[user_name]}}' failed.<br>\n<br>\n User was not found,
```
Comment (mrinehart-r7) [inline workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams.icon:684]:
```suggestion
                "message_content": "⚠️ Disable Active Directory User request for '{{[a4934e67-ceaa-4eda-9b1f-c023175b37e9].[user_name]}}' failed.<br>\n<br>\n User was not found,
```

File: workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams.icon [hunk 7]
```diff
@@ -0,0 +1,1379 @@
+{
+  "kom": {
+    "komandVersion": "1.49.2-0-gafb52",
+    "komFileVersion": "2.0.0",
+    "exportedAt": "2020-04-22T18:37:22.994611519Z",
+    "workflowVersions": [
+      {
+        "name": "Disable User in Microsoft Active Directory with Microsoft Teams",
```
Comment (jschipp-r7) [inline workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams.icon:8]:
```suggestion
        "name": "Disable User in Active Directory with Microsoft Teams",
```

File: workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/help.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+Disable User in Microsoft Active Directory with Microsoft Teams.
+
+Sample Slack Trigger Commands:
+
+`!disable_user <exampleuser>`
+
+# Key Features
+
+* Disable an Active Directory LDAP user directly from Microsoft Teams
```
Comment (jschipp-r7) [inline workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/help.md:11]:
```suggestion
* Disable an Active Directory user directly from Microsoft Teams
```

File: workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/help.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+# Requirements
+
+* [Microsoft Teams](https://insightconnect.help.rapid7.com/docs/microsoft-teams)
+* Active Directory LDAP connection
```
Comment (jschipp-r7) [inline workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/help.md:16]:
```suggestion
* Active Directory connection
```

File: workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/help.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+# Documentation
+
+## Setup
+
+Import the workflow from the Rapid7 Extension Library and proceed through the Import Workflow wizard in InsightConnect. Import plugins, create or select connections, and rename the workflow as a part of the Import Workflow wizard as necessary.
+
+Once the workflow has been imported, each Microsoft Teams step will need the team name and channel name updated to suit your Teams environment (edit the input with the preset text of `change_me`).
+
+To run the workflow, send the command `!disable_user <exampleuser>` in your configured team and channel. The workflow will post after it has completed.
+In the Team and chanel inputs replace `change_me` with the appropriate team and channel.
```
Comment (jschipp-r7) [inline workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/help.md:27]:
```suggestion
In the Team and channel inputs replace `change_me` with the appropriate team and channel.
```

File: workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/workflow.spec.yaml [hunk 11]
```diff
@@ -0,0 +1,29 @@
+extension: workflow
+products: ["insightconnect"]
+name: Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams
+title: "Disable User in Microsoft Active Directory with Microsoft Teams"
```
Comment (jschipp-r7) [inline workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/workflow.spec.yaml:4]:
```suggestion
title: "Disable User in Active Directory from Microsoft Teams"
```

File: workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/workflow.spec.yaml [hunk 12]
```diff
@@ -0,0 +1,29 @@
+extension: workflow
+products: ["insightconnect"]
+name: Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams
```
Comment (jschipp-r7) [inline workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/workflow.spec.yaml:3]:
```suggestion
name: Disable_User_in_Active_Directory_from_Microsoft_Teams
```

File: workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/workflow.spec.yaml [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+title: "Disable User in Microsoft Active Directory with Microsoft Teams"
+description: Disable Active Directory LDAP users from Slack.
```
Comment (jschipp-r7) [inline workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/workflow.spec.yaml:5]:
```suggestion
description: Disable Active Directory users from Slack.
```

File: workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/workflow.spec.yaml [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+version: 1.0.0
+vendor: rapid7
+support: rapid7
+status: []
+hub_tags:
+  use_cases: [threat_detection_and_response, cloud_security, user_management]
+  keywords: [chat, active_directory, ldap, teams, work_from_home, phishing, covid]
```
Comment (jschipp-r7) [inline workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/workflow.spec.yaml:12]:
```suggestion
  keywords: [chat, active_directory, ldap, teams, work_from_home, phishing]
```

File: workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/workflow.spec.yaml [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+  features: []
+resources:
+  source_url: https://github.com/rapid7/insightconnect-workflows/tree/master/workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams
```
Comment (jschipp-r7) [inline workflows/Disable_User_in_Microsoft_Active_Directory_with_Microsoft_Teams/workflow.spec.yaml:15]:
```suggestion
  source_url: https://github.com/rapid7/insightconnect-workflows/tree/master/workflows/Disable_User_in_Active_Directory_from_Microsoft_Teams
```

File: workflows/Disable_User_in_Active_Directory_from_Microsoft_Teams/workflow.spec.yaml [hunk 16]
```diff
@@ -0,0 +1,29 @@
+extension: workflow
+products: ["insightconnect"]
+name: Disable_User_in_Active_Directory_from_Microsoft_Teams
+title: "Disable User in  Active Directory from Microsoft Teams"
```
Comment (jschipp-r7) [inline workflows/Disable_User_in_Active_Directory_from_Microsoft_Teams/workflow.spec.yaml:4]:
```suggestion
title: "Disable User in Active Directory from Microsoft Teams"
```

File: workflows/Disable_User_in_Active_Directory_from_Microsoft_Teams/workflow.spec.yaml [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+products: ["insightconnect"]
+name: Disable_User_in_Active_Directory_from_Microsoft_Teams
+title: "Disable User in Active Directory from Microsoft Teams"
+description: Disable Active Directory users from Microsoft Teams.
+version: 1.0.0
+vendor: rapid7
+support: rapid7
+status: []
+hub_tags:
+  use_cases: [threat_detection_and_response, cloud_security, user_management]
```
Comment (jschipp-r7) [inline workflows/Disable_User_in_Active_Directory_from_Microsoft_Teams/workflow.spec.yaml:11]:
```suggestion
  use_cases: [threat_detection_and_response, user_management]
```

File: workflows/Disable_User_in_Active_Directory_from_Microsoft_Teams/help.md [hunk 18]
```diff
@@ -0,0 +1,51 @@
+# Description
+
+Disable User in Active Directory from Microsoft Teams.
```
Comment (sengleson-r7) [inline workflows/Disable_User_in_Active_Directory_from_Microsoft_Teams/help.md:3]:
```suggestion
Disabling a compromised account can limit the scope of an attack and buy valuable time to investigate and contain the threat. This workflow disables a domain user account with a command in Microsoft Teams.
```

File: workflows/Disable_User_in_Active_Directory_from_Microsoft_Teams/help.md [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+
+Sample Slack Trigger Commands:
+
+`!disable-user <exampleuser>`
+
+# Key Features
+
+* Disable an Active Directory user directly from Microsoft Teams
```
Comment (sengleson-r7) [inline workflows/Disable_User_in_Active_Directory_from_Microsoft_Teams/help.md:11]:
```suggestion
* **Break the Kill Chain** - Credentials provide attackers with easy access to numerous targets. Disabling a compromised account or forcing a password reset can quickly and effectively interrupt an attacker’s kill chain.
* **Buy Time to Investigate and Remediate** - In a pinch, disabling a user account can limit your threat exposure and buy your team valuable time to investigate and contain a threat. 
* **Reduce Portal Fatigue** - You have enough to do without logging into a different solution every 5 minutes. Control your response actions from chat instead.
```

File: workflows/Disable_User_in_Active_Directory_from_Microsoft_Teams/help.md [hunk 20]
```diff
@@ -0,0 +1,51 @@
...
+# Documentation
+
+## Setup
+
+Import the workflow from the Rapid7 Extension Library and proceed through the Import Workflow wizard in InsightConnect. Import plugins, create or select connections, and rename the workflow as a part of the Import Workflow wizard as necessary.
+
+Once the workflow has been imported, each Microsoft Teams step will need the team name and channel name updated to suit your Teams environment (edit the input from the preset text of `change_me`).
+
+To run the workflow, send the command `!disable-user <exampleuser>` in your configured team and channel. The workflow will post after it has completed.
+In the Team and channel inputs replace `change_me` with the appropriate team and channel.
```
Comment (sengleson-r7) [inline workflows/Disable_User_in_Active_Directory_from_Microsoft_Teams/help.md:27]:
```suggestion
Once the workflow has been imported, **each Microsoft Teams step will need the team name and channel name updated to suit your Teams environment!** Edit the input with the preset text of `change_me` in each Teams step in the workflow.

After configuring the Teams steps, activate the workflow in order to trigger it.

# Usage

*This workflow will only trigger in the channel specified in the Microsoft Teams workflow steps.*

To run the workflow, send a message to the specified Microsoft Teams channel starting with the command `!disable-user`.

Commands should be in the following format:

`!disable-user jdoe`

The workflow will post the results in a thread.
```

**Pull request merged**